### PR TITLE
Add WithHostPid func to pod.go

### PR DESCRIPTION
**Pull request by shaior**

### CODE REVIEW

File: pkg/pod/pod.go [hunk 1]
```diff
@@ -724,6 +724,25 @@ func (builder *Builder) WithHostNetwork() *Builder {
 	return builder
 }
 
+// WithHostPid allows a pod to have access to the host process ID namespace.
+func (builder *Builder) WithHostPid() *Builder {
```
Comment (kononovn) [inline pkg/pod/pod.go:728]: Add boolean flag as a parameter. May be in some cases you'll need explicitly false flag.

File: pkg/pod/pod.go [hunk 2]
```diff
@@ -724,6 +724,25 @@ func (builder *Builder) WithHostNetwork() *Builder {
...
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof("Applying HostPID flag to pod's %s configuration", builder.Definition.Name)
```
Comment (achuzhoy) [inline pkg/pod/pod.go:733]: nit: if the pod name is "sun" - then this reads:
"Applying HostPID flag to pod's sun configuration"  which can be confusing.

Suggest to change:
"Applying HostPID flag to the configuration of pod %s"

Comment (trewest) [reply pkg/pod/pod.go:733]: Please also include the namespace that the pod is running in

File: pkg/pod/pod.go [hunk 3]
```diff
@@ -724,6 +724,26 @@ func (builder *Builder) WithHostNetwork() *Builder {
 	return builder
 }
 
+// WithHostPid configures a pod's access to the host process ID namespace based on a boolean parameter.
+func (builder *Builder) WithHostPid(flag bool) *Builder {
```
Comment (kononovn) [inline pkg/pod/pod.go:728]: Please re-name argument from "flag" to simple "hostPid"

Comment (shaior) [reply pkg/pod/pod.go]: done.

**Review by kononovn — APPROVED** [review]
/lgtm

File: pkg/pod/pod.go [hunk 4]
```diff
@@ -724,6 +724,26 @@ func (builder *Builder) WithHostNetwork() *Builder {
 	return builder
 }
 
+// WithHostPid configures a pod's access to the host process ID namespace based on a boolean parameter.
+func (builder *Builder) WithHostPid(hostPid bool) *Builder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof("Applying HostPID flag to the configuration of pod: %s in namespace: %s:",
```
Comment (trewest) [inline pkg/pod/pod.go:733]: nit
```suggestion
	glog.V(100).Infof("Applying HostPID flag to the configuration of pod: %s in namespace: %s",
```

**Review by trewest — CHANGES_REQUESTED** [review]
Small nit, otherwise lgtm

**Review by trewest — APPROVED** [review]
/lgtm

**Pull request merged**